frontend/src/util/activity.js

```diff
@@ -0,0 +1,191 @@
+import {
+  TLM_ENTITY_TYPE as TLM_ET,
+  TLM_SUB_TYPE as TLM_ST,
+  getContentComment,
+  handleFetchResult,
+  getWorkspaceContent
+} from 'tracim_frontend_lib'
+
+const createActivityEvent = (message) => {
+  return {
+    eventId: message.event_id,
+    eventType: message.event_type,
+    author: message.fields.author,
+    created: message.created
+  }
+}
+
+const createMemberActivity = (activityId, messageList) => {
+  if (messageList.length !== 1) throw new Error('Must have at exactly one message to build a member activity')
+  const message = messageList[0]
+  return {
+    id: activityId,
+    entityType: message.event_type.split('.')[0],
+    eventList: [createActivityEvent(message)],
+    reactionList: [],
+    newestMessage: message
+  }
+}
+
+const createSubscriptionActivity = (activityId, messageList) => {
+  if (messageList.length !== 1) throw new Error('Must have at exactly one message to build a subscription activity')
+  const message = messageList[0]
+  return {
+    id: activityId,
+    entityType: message.event_type.split('.')[0],
+    eventList: [createActivityEvent(message)],
+    reactionList: [],
+    newestMessage: message
+  }
+}
+
+// INFO - SG - 2020-11-12 - this function assumes that the list is ordered from newest to oldest
+const createContentActivity = async (activityId, messageList, apiUrl) => {
+  if (!messageList.length) throw new Error('Must have at least one message to build a content activity')
+  const first = messageList[0]
+
+  let content = first.fields.content
+  if (content.content_type === TLM_ST.COMMENT) {
+    const response = await handleFetchResult(await getWorkspaceContent(
+      apiUrl,
+      first.fields.workspace.workspace_id,
+      content.parent_content_type + 's',
+      content.parent_id
+    ))
+    if (response.apiResponse.status === 200) {
+      content = response.body
+    }
+  }
+
+  const response = await handleFetchResult(await getContentComment(apiUrl, content.workspace_id, content.content_id))
+  const commentList = response.apiResponse.status === 200 ? response.body : []
+  return {
+    id: activityId,
+    entityType: first.event_type.split('.')[0],
+    eventList: messageList.map(createActivityEvent),
+    reactionList: [],
+    commentList: commentList,
+    newestMessage: first,
+    content: content
+  }
+}
+
+const getActivityId = (message) => {
+  const [entityType, , subEntityType] = message.event_type.split('.')
+  let id = null
+  switch (entityType) {
+    case TLM_ET.CONTENT:
+    case TLM_ET.MENTION:
+      id = (subEntityType === TLM_ST.COMMENT)
```

Comment:
A comment's activity is the activity of its content, so I think the `parent_id` is OK for comments. Using `content_id` would lead to display an activity for a comment.

Reply:
sorry, I think my comment was not clear, my question is: For the TLM_ET.MENTION we should look to the parent_id too, but I don't think that `(subEntityType === TLM_ST.COMMENT)` will be true in that case

Reply:
For `mention.created` the `content` field really contains the "parent" content as a mention is not a content. Like I wrote in another answer I agree this is not intuitive at all, I (now) think it would have been better to have a `comment.created` tlm with a `content` field containing its parent rather than the `content.created.comment` we have currently.